File: java/src/main/java/QuestionNine.java
import java.util.Collections;
import java.util.LinkedList;
import java.util.Queue;

// Breadth First Search approach //
public class QuestionNine {
	public static int sum(Node root) {
		int sum = 0;
		if (root == null) return 0;
		Queue<Node> queue = new LinkedList<>();
		queue.add(root);
		while (!queue.isEmpty()) {
			int size = queue.size();
			for (int i = 0; i < size; i++) {
				Node node = queue.poll();
				sum += node.val;
				if (node.children != null) Collections.addAll(queue, node.children);
			}
		}
		return sum;
	}
}

File: go/questions/question_nine.go
package questions

import (
	"container/list"
)

type Node struct {
	Val      int
	Children []Node
}

// Sum Breadth First Search approach //
func Sum(root Node) int {
	sum := 0
	queue := list.New()
	queue.PushBack(root)
	for queue.Len() > 0 {
		size := queue.Len()
		for i := 0; i < size; i++ {
			node := getNodeFromQueueElement(queue.Front().Value)
			queue.Remove(queue.Front())
			sum += node.Val
			if node.Children != nil {
				children := node.Children
				for _, child := range children {
					queue.PushBack(child)
				}
			}
		}
	}
	return sum
}

func getNodeFromQueueElement(element any) Node {
	if node, ok := element.(Node); ok {
		return node
	} else {
		return Node{}
	}
}
